server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2286,6 +2276,23 @@ protected SegmentInfos getLastCommittedSegmentInfos() {
         return lastCommittedSegmentInfos;
     }
 
+    @Override
+    public SegmentInfos getLatestSegmentInfos() {
+        OpenSearchDirectoryReader reader = null;
+        try {
+            reader = externalReaderManager.internalReaderManager.acquire();
```

Comment:
You could use just `internalReaderManager` (https://github.com/opensearch-project/OpenSearch/pull/3240/files#diff-9d23f98d69ed60603d7dd374871a9e86ad5b176ca105f842f4adad7a7c153a50L295)